HLTrigger/Timer/interface/ProcessCallGraph.h

```diff
@@ -0,0 +1,193 @@
+#include "HLTrigger/Timer/interface/EDMModuleType.h"
+
+class ProcessCallGraph {
+public:
+
+  struct node_type {
+    edm::ModuleDescription module_;
+    edm::EDMModuleType     type_;
+    bool                   scheduled_;
+  };
+
+  // directed graph, with `node_type` properties attached to each vertex
+  using graph_type = boost::subgraph<boost::adjacency_list<
+      // edge list
+      boost::vecS,
+      // vertex list
+      boost::vecS,
+      boost::directedS,
+      // vertex properties
+      node_type,
+      // edge propoerties, used internally by boost::subgraph
+      boost::property<boost::edge_index_t, int>,
+      // graph properties, used to name each boost::subgraph
+      boost::property<boost::graph_name_t, std::string>
+  >>;
+
+  // store the details of each path: name, modules on the path, and their dependencies
+  struct path_type {
+    std::string                 name_;
+    std::vector<unsigned int>   modules_on_path_;
+    std::vector<unsigned int>   modules_and_dependencies_;
+    std::vector<unsigned int>   last_dependency_of_module_;   // one-after-the-last dependency of each module, as indices into modules_and_dependencies_
+
+    path_type() = default;
+
+    path_type(std::string const & name, std::vector<unsigned int> const & mop, std::vector<unsigned int> const & mad, std::vector<unsigned int> const & ldom) :
+      name_(name),
+      modules_on_path_(mop),
+      modules_and_dependencies_(mad),
+      last_dependency_of_module_(ldom)
+    { }
+
+    path_type(std::string && name, std::vector<unsigned int> && mop, std::vector<unsigned int> && mad, std::vector<unsigned int> && ldom) :
+      name_(std::move(name)),
+      modules_on_path_(std::move(mop)),
+      modules_and_dependencies_(std::move(mad)),
+      last_dependency_of_module_(std::move(ldom))
+    { }
+
+    path_type(path_type const & other) = default;
+    
+    path_type(path_type && other) = default;
+
+    ~path_type() = default;
+
+    path_type & operator=(path_type const & other) = default;
+
+  };
+
+  // store the details of each process: name, modules call subgraph, modules, paths and endpaths, subprocess pids
+  struct process_type {
+    std::string                 name_;
+    graph_type const &          graph_;
+    std::vector<unsigned int>   modules_;
+    std::vector<path_type>      paths_;
+    std::vector<path_type>      endPaths_;
+    std::vector<unsigned int>   subprocesses_;
+
+    process_type() = delete;
+
+    process_type(
+      std::string const & name,
+      graph_type const & graph,
+      std::vector<unsigned int> const & modules,
+      std::vector<path_type> const & paths,
+      std::vector<path_type> const & endPaths,
+      std::vector<unsigned int> const & subprocesses = {}
+    ) :
+      name_(name),
+      graph_(graph),
+      modules_(modules),
+      paths_(paths),
+      endPaths_(endPaths),
+      subprocesses_(subprocesses)
+    { }
+
+    process_type(
+      std::string && name,
```

Comment:
You do not want to use `&&`, instead just do `std::string`. It will work the exact same way.